src/client/extension.ts

```diff
@@ -104,24 +104,31 @@ async function activateUnsafe(
     startupDurations: Record<string, number>
 ): Promise<[IExtensionApi, Promise<void>, IServiceContainer]> {
     const activationDeferred = createDeferred<void>();
-    displayProgress(activationDeferred.promise);
-    startupDurations.startActivateTime = startupStopWatch.elapsedTime;
+    try {
+        displayProgress(activationDeferred.promise);
+        startupDurations.startActivateTime = startupStopWatch.elapsedTime;
 
-    //===============================================
-    // activation starts here
+        //===============================================
+        // activation starts here
 
-    const [serviceManager, serviceContainer] = initializeGlobals(context);
-    activatedServiceContainer = serviceContainer;
-    const { activationPromise } = await activateComponents(context, serviceManager, serviceContainer);
+        const [serviceManager, serviceContainer] = initializeGlobals(context);
+        activatedServiceContainer = serviceContainer;
+        const { activationPromise } = await activateComponents(context, serviceManager, serviceContainer);
 
-    //===============================================
-    // activation ends here
+        //===============================================
+        // activation ends here
 
-    startupDurations.endActivateTime = startupStopWatch.elapsedTime;
-    activationDeferred.resolve();
+        startupDurations.endActivateTime = startupStopWatch.elapsedTime;
+        activationDeferred.resolve();
 
-    const api = buildApi(activationPromise, serviceManager, serviceContainer);
-    return [api, activationPromise, serviceContainer];
+        const api = buildApi(activationPromise, serviceManager, serviceContainer);
+        return [api, activationPromise, serviceContainer];
+    } finally {
+        // Make sure that we clear our status message
+        if (!activationDeferred.completed) {
```

Comment:
Found this while validating 5145 fix. We could allow python to load, but we left up our loading message:
![image](https://user-images.githubusercontent.com/812783/112065495-f2493e00-8b21-11eb-86aa-645793dfebef.png)
Jupyter Extension loading would never go away.

Reply:
If we reject here, does VS Code know that our activation has failed, and would this cause extensions depending on us to also fail to activate?

Reply:
Double checking now. But I believe this reject is safe to use. This reject is only on the activationDeferred which we only use for turning on or off the window.withProgress status message. The window.withProgress is just a noop on the then, so it doesn't care if the deferred is resolved via resolve or reject, just that it's done. 

The actual error thrown takes the same route as before and is handled up at line 65 of this file where the user is notified and the "fake" api is returned to allow Python to function even if we fail to load. Don already made that fix before, this was just for the status message not being cleared in that scenario.